enginerpc/engine_impl.cc

```diff
@@ -168,20 +169,26 @@ grpc::Status engine_impl::GetHost(grpc::ServerContext* context
             auto ithostname = host::hosts.find(request->name());
             if (ithostname != host::hosts.end())
               selectedhost = ithostname->second;
-            else
+            else {
+              err = "could not find host";
```

Comment:
Could you be more precise, something like
err = fmt::format("could not find host '{}'", request->name());
would look better.